lib/field.dart

```diff
@@ -35,36 +44,51 @@ class _FieldWidgetState extends State<FieldWidget> {
     return "FieldWidget ${widget.index}, outgoingArrows: $outgoingArrows, incomingArrows: $incomingArrows";
   }
 
+  final GlobalKey _globalKey = GlobalKey();
+
   @override
   Widget build(BuildContext context) {
     var controller = widget.gridController;
-    return GestureDetector(
+    return InkWell(
+      key: _globalKey,
       onTap: () {
         if (controller.bottomBarState == null) return;
 
+        RenderBox box =
+            _globalKey.currentContext!.findRenderObject() as RenderBox;
+        Offset position =
+            box.localToGlobal(const Offset(35, -50)); //TODO Arrow Postion
+
+        ArrowData currentIndexData =
+            ArrowData(arrowIndex: widget.index, arrowOffset: position);
+
         switch (controller.bottomBarState!) {
           case Items.arrow:
             if (controller.creatingArrow) {
-              controller.arrowEnd = widget.index;
+              // Edit Essam Get Offset
+
+              controller.arrowEnd = currentIndexData;
 
               assert(controller.arrowStart != null);
-              assert(controller.onArrowCreated != null);
+              assert(controller.onArrowCreated != null); // Remove it
 
               // make sure the arrow is not already created
               if (!incomingArrows.contains(controller.arrowStart)) {
                 incomingArrows.add(controller.arrowStart!);
 
                 print(
-                    "Arrow created from ${controller.arrowStart} to ${controller.arrowEnd}");
-                //inform the source field of the arrow creation.
-                controller.onArrowCreated!(widget.index);
+                    "Arrow created from ${controller.arrowStart?.arrowIndex} to ${controller.arrowEnd?.arrowIndex}");
+
+                controller.onArrowCreated!(currentIndexData); // Remove it
```

Comment:
if this is removed, the outgoingArrows in the start field will not get populated, right?